File: generation-scripts/seed-supabase.py
#!/usr/bin/env python3
"""
Seed Supabase database with all Templata content from TypeScript files.
"""

import os
import json
import re
from supabase import create_client, Client

# Supabase configuration
SUPABASE_URL = "https://uvcstcajctqbxddosdbf.supabase.co"
SUPABASE_KEY = "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJzdXBhYmFzZSIsInJlZiI6InV2Y3N0Y2FqY3RxYnhkZG9zZGJmIiwicm9sZSI6InNlcnZpY2Vfcm9sZSIsImlhdCI6MTc1Nzk3OTcwNywiZXhwIjoyMDczNTU1NzA3fQ.PjcGj6uFV1M5yFju1YYvbxG3ld55llauMUPlrjow6I4"

# Initialize Supabase client
supabase: Client = create_client(SUPABASE_URL, SUPABASE_KEY)

def parse_ts_file_to_dict(filepath):
    """Parse TypeScript file and extract data as dictionary"""
    with open(filepath, 'r', encoding='utf-8') as f:
        content = f.read()

    # Find the exported array/object
    # This is a simplified parser - assumes specific format
    data_match = re.search(r'export const \w+ = ({.*?});', content, re.DOTALL)
    if not data_match:
        data_match = re.search(r'export const \w+: \w+\[\] = (\[.*?\]);', content, re.DOTALL)

    if data_match:
        # This is hacky but works for our specific format
        # In production, you'd use a proper TS parser
        data_str = data_match.group(1)
        # Convert TS to JSON-like format (this is simplified)
        # For now, we'll use exec to evaluate it
        return None  # We'll handle this differently

    return None

def seed_articles():
    """Seed articles from manifest"""
    print("Seeding articles...")

    with open('../src/data/articles/manifest.json', 'r') as f:
        manifest = json.load(f)

    batch = []
    count = 0

    for item in manifest:
        filepath = f"../src/data/articles/{item['filename']}.ts"

        # Read the file and extract article data
        # For now, we'll skip parsing and just track progress
        # In next iteration, we'll properly parse the TS files

        count += 1
        if count % 1000 == 0:
            print(f"  Processed {count}/{len(manifest)} articles...")

    print(f"✓ Would seed {count} articles (parser implementation pending)")
    return count

def seed_prompts():
    """Seed prompts from manifest"""
    print("Seeding prompts...")

    with open('../src/data/prompts/manifest.json', 'r') as f:
        manifest = json.load(f)

    print(f"✓ Would seed {len(manifest)} prompts (parser implementation pending)")
    return len(manifest)

def seed_marketing():
    """Seed marketing pages from manifest"""
    print("Seeding marketing pages...")

    with open('../src/data/marketing/manifest.json', 'r') as f:
        manifest = json.load(f)

    print(f"✓ Would seed {len(manifest)} marketing pages (parser implementation pending)")
    return len(manifest)

def seed_templates():
    """Seed templates from manifest"""
    print("Seeding templates...")

    with open('../src/data/templates/manifest.json', 'r') as f:
        manifest = json.load(f)

    print(f"✓ Would seed {len(manifest)} templates (parser implementation pending)")
    return len(manifest)

if __name__ == '__main__':
    print("=" * 60)
    print("Templata Supabase Seeder")
    print("=" * 60)
    print()

    # Test connection
    try:
        result = supabase.table('templata_articles').select('count').execute()
        print("✓ Connected to Supabase successfully")
        print()
    except Exception as e:
        print(f"✗ Failed to connect to Supabase: {e}")
        print("  Make sure you've run the schema.sql file first!")
        exit(1)

    articles = seed_articles()
    prompts = seed_prompts()
    marketing = seed_marketing()
    templates = seed_templates()

    print()
    print("=" * 60)
    print(f"Summary: {articles + prompts + marketing + templates} total items")
    print("=" * 60)
    print()
    print("NOTE: This is a dry-run. Actual TS parser implementation needed.")
    print("Next step: Install supabase-py and implement proper TS parsing.")

File: scripts/seed-supabase.ts
#!/usr/bin/env tsx
/**
 * Seed Supabase database with all Templata content
 * Run with: npx tsx scripts/seed-supabase.ts
 */

import { createClient } from '@supabase/supabase-js';
import { readdir } from 'fs/promises';
import { join } from 'path';

const SUPABASE_URL = process.env.NEXT_PUBLIC_SUPABASE_URL!;
const SUPABASE_KEY = process.env.SUPABASE_SERVICE_ROLE_KEY!;

const supabase = createClient(SUPABASE_URL, SUPABASE_KEY);

async function seedArticles() {
  console.log('Seeding articles...');

  const manifestPath = join(process.cwd(), 'src/data/articles/manifest.json');
  const manifest = await import(manifestPath);

  let count = 0;
  let batch: any[] = [];
  const BATCH_SIZE = 100;

  for (const item of manifest.default) {
    try {
      const modulePath = `../src/data/articles/${item.filename}`;
      const module = await import(modulePath);

      if (module.articles && module.articles.length > 0) {
        for (const article of module.articles) {
          batch.push({
            id: article.id,
            title: article.title,
            excerpt: article.excerpt,
            content: article.content,
            author: article.author,
            published_at: article.publishedAt,
            updated_at: article.updatedAt,
            read_time: article.readTime,
            category: article.category,
            featured: article.featured || false,
            tags: article.tags,
            slug: article.slug,
            type: article.type,
            difficulty: article.difficulty,
            meta_title: article.seo?.metaTitle,
            meta_description: article.seo?.metaDescription,
            og_image: article.seo?.ogImage,
            related_templates: article.relatedTemplates || [],
            related_posts: article.relatedPosts || []
          });

          count++;

          if (batch.length >= BATCH_SIZE) {
            const { error } = await supabase
              .from('templata_articles')
              .upsert(batch);

            if (error) {
              console.error('Error inserting batch:', error);
            } else {
              console.log(`  Inserted ${count} articles...`);
            }

            batch = [];
          }
        }
      }
    } catch (error) {
      console.error(`Error processing ${item.filename}:`, error);
    }
  }

  // Insert remaining batch
  if (batch.length > 0) {
    const { error } = await supabase
      .from('templata_articles')
      .upsert(batch);

    if (error) {
      console.error('Error inserting final batch:', error);
    }
  }

  console.log(`✓ Seeded ${count} articles`);
  return count;
}

async function seedPrompts() {
  console.log('Seeding prompts...');

  const manifestPath = join(process.cwd(), 'src/data/prompts/manifest.json');
  const manifest = await import(manifestPath);

  let count = 0;
  let batch: any[] = [];
  const BATCH_SIZE = 500;

  for (const item of manifest.default) {
    try {
      const modulePath = `../src/data/prompts/${item.filename}`;
      const module = await import(modulePath);

      if (module.prompts && module.prompts.length > 0) {
        for (const prompt of module.prompts) {
          batch.push({
            id: prompt.id,
            prompt: prompt.prompt,
            category: prompt.category,
            type: prompt.type,
            template_id: item.template
          });

          count++;

          if (batch.length >= BATCH_SIZE) {
            const { error } = await supabase
              .from('templata_prompts')
              .upsert(batch);

            if (error) {
              console.error('Error inserting batch:', error);
            } else {
              console.log(`  Inserted ${count} prompts...`);
            }

            batch = [];
          }
        }
      }
    } catch (error) {
      console.error(`Error processing ${item.filename}:`, error);
    }
  }

  // Insert remaining batch
  if (batch.length > 0) {
    const { error} = await supabase
      .from('templata_prompts')
      .upsert(batch);

    if (error) {
      console.error('Error inserting final batch:', error);
    }
  }

  console.log(`✓ Seeded ${count} prompts`);
  return count;
}

async function main() {
  console.log('='.repeat(60));
  console.log('Templata Supabase Seeder');
  console.log('='.repeat(60));
  console.log('');

  // Test connection
  try {
    const { data, error } = await supabase
      .from('templata_articles')
      .select('count')
      .limit(1);

    if (error) throw error;
    console.log('✓ Connected to Supabase successfully');
    console.log('');
  } catch (error) {
    console.error('✗ Failed to connect to Supabase:', error);
    console.log('  Make sure you have run the schema.sql file first!');
    process.exit(1);
  }

  const articles = await seedArticles();
  const prompts = await seedPrompts();

  console.log('');
  console.log('='.repeat(60));
  console.log(`Summary: ${articles + prompts} total items seeded`);
  console.log('='.repeat(60));
}

main().catch(console.error);
